src/cache-restore.ts

```diff
@@ -0,0 +1,57 @@
+import * as cache from '@actions/cache';
+import * as core from '@actions/core';
+import * as glob from '@actions/glob';
+import path from 'path';
+import fs from 'fs';
+
+import {State, Outputs} from './constants';
+import {
+  getCacheDirectoryPath,
+  getPackageManagerInfo,
+  PackageManagerInfo
+} from './cache-utils';
+
+export const restoreCache = async (packageManager: string) => {
+  const packageManagerInfo = await getPackageManagerInfo(packageManager);
+  if (!packageManagerInfo) {
+    throw new Error(`Caching for '${packageManager}' is not supported`);
+  }
+  const platform = process.env.RUNNER_OS;
+
+  const cachePath = await getCacheDirectoryPath(
+    packageManagerInfo,
+    packageManager
+  );
+  const lockFilePath = findLockFile(packageManagerInfo);
+  const fileHash = await glob.hashFiles(lockFilePath);
+
+  const primaryKey = `${platform}-${packageManager}-${fileHash}`;
```

Comment:
May be we can add `node-cache` prefix here?
`node-cache-${platform}-${packageManager}-${fileHash}`

Reply:
Done. Thank you.